app/controllers/toezicht/inzendingen.js

```diff
@@ -18,22 +23,47 @@ export default Controller.extend({
     this.set('header', ENV['vo-webuniversum']['header']);
   },
 
-  filterChanged: observer('bestuurseenheidId', 'classificatieId', 'provincieId', 'besluitTypeId',
-                          'sessionDateFrom', 'sessionDateTo', 'sentDateFrom', 'sentDateTo', 'statusId', function() {
+  filterChanged: observer('bestuurseenheidId', 'classificatieId', 'provincieId',
+                          'besluitType', 'toezichtRegulationTypeId', 'sessionDateFrom', 'sessionDateTo', 'sentDateFrom', 'sentDateTo', 'statusUri', function() {
     this.set('page', 0);
   }),
 
+  isRegulation: computed('besluitType.isRegulation', function(){
+    if (this.besluitType === undefined && this.besluitTypeId !== undefined)
+      this.set('besluitType', this.store.findRecord('besluitType', this.besluitTypeId));
```

Comment:
What you want to achieve here is to lookup the `besluit-type` record here for a given `besluitTypeId`. This logic already happens in the `didReceiveAttrs` hook of the `besluit-type-select` component, but currently other parts of the application aren't notified about the resolved value by the `besluit-type-select` component. Therefore, I would implement an `onInit(besluitType)` hook on that component that is called in the `didReceiveAttrs` with the resolved value.
Consumers of the `besluit-type-select` component (like this `inzendingen` route) can then implement the `onInit` hook to do something with the initial value of `besluitType`.

Reply:
Yes, this is the correct approach and the onInit hook was already there.

Reply:
You still need to call the `onInit` method at the side of the `besluit-type-select` component with the resolved `besluitType` value as an argument.

Reply:
Would you please let me know on which line you suppose the call shall be placed? Because it apparently gets called automatically. At first I had called onInit in the didReceivettrs hook, but it complained that you are modifying a value twice.

Reply:
That's the correct place to put it.
The error appears because you pass the `readBesluitType`action to the `onInit` hook which updates `besluitType` and `besluitTypeId`. You should define a new action that only sets the `besluitType`. That can be even done inline using the `mut` helper.